crates/fuel-core/src/database.rs

```diff
@@ -66,25 +74,26 @@ use std::path::Path;
 use tempfile::TempDir;
 
 // Storages implementation
-mod block;
-mod contracts;
-mod message;
-mod sealed_block;
-mod state;
-
-pub(crate) mod coin;
-
 pub mod balances;
+pub mod block;
+pub mod coin;
+pub mod contracts;
+pub mod database_description;
+pub mod message;
 pub mod metadata;
+pub mod sealed_block;
+pub mod state;
+pub mod statistic;
 pub mod storage;
 pub mod transaction;
 pub mod transactions;
 
-pub type Column = fuel_core_storage::column::Column;
-
 #[derive(Clone, Debug)]
-pub struct Database {
-    data: StructuredStorage<DataSource>,
+pub struct Database<Description = OnChain>
```

Comment:
Why be opinionated about a default value for `Description`?

Reply:
I did it to minimize the number of changes -> conflicts with regenesis PR